### ISSUE TITLE
No longer working in FF 80+

### ISSUE THREAD

**Issue by ornory** [ufb/Textmarker#147]
Thanks

**Comment by giammi56** [comment]
Same problem here. Which information should I provide to be helpful?

**Comment by cheeberger** [comment]
Same here. I swear this is why I don't update sometimes.

11/12/2020, 09:26:28 AM | Unable to save your highlights.
11/12/2020, 09:15:27 AM | Unable to save your highlights.
11/12/2020, 09:15:11 AM | Unable to save your highlights.
11/12/2020, 09:14:53 AM | Unable to save your highlights.
windows:1809
ff: 82.0.3

**Comment by sigfriedCub1990** [comment]
I think this project is abandoned :(. That's sad

**Comment by ufb** [comment]
Hi y'all,
not abandoned, but 2020 was a crazy year and I just didn't find the time to investigate this issue further, after having unsuccessfully asked Mozilla for help, so I wasn't able to get to the bottom of it, why Textmarker stopped working on FF updates for some users.
I can't tell when or even if this is going to be fixed, since I don't even know the root of the problem. I checked Mozilla's developer notes on FF update changes, but that wasn't very enlightening either -> communication between extension devs and Mozilla's addon team is almost non-existent, I can't even send them an e-mail, all I could do is posting an issue in their forum, but that wasn't of any help.
Sorry for the inconvenience!

**Comment by ufb** [comment]
This seems to be the same issue as this one: #144 -> the problem seems to be related to FF updates (not to particular FF versions)

**Comment by sigfriedCub1990** [comment]
> Hi y'all,
> not abandoned, but 2020 was a crazy year and I just didn't find the time to investigate this issue further, after having unsuccessfully asked Mozilla for help, so I wasn't able to get to the bottom of it, why Textmarker stopped working on FF updates for some users.
> I can't tell when or even if this is going to be fixed, since I don't even know the root of the problem. I checked Mozilla's developer notes on FF update changes, but that wasn't very enlightening either -> communication between extension devs and Mozilla's addon team is almost non-existent, I can't even send them an e-mail, all I could do is posting an issue in their forum, but that wasn't of any help.
> Sorry for the inconvenience!

That's very good to hear, sorry for drawing rushed conclusions about it. Yesterday I was inspecting the code to see if I can resolve the issue, this is a very helpful extension so I would try my best to solve the problem. However, can you @ufb provide some guidance with the codebase? Thanks in advance. Cheers, sigfried.

**Comment by ufb** [comment]
@sigfriedCub1990 

> can you @ufb provide some guidance with the codebase?

Of course!
I just wrote you an email to the address from your GitHub profile page. You can ask your questions via email and I'll try my best to answer them.

**Comment by ufb** [comment]
@sigfriedCub1990, btw: It's possible that the issue is related to some FF privacy (or other) changes with respect to synced storage - at least that was some user's hint I got some weeks ago.

**Comment by sigfriedCub1990** [comment]
@ufb, yes, that could be the cause. Yesterday I was looking at some similar error messages and some of them were related to FF permissions.

**Comment by sigfriedCub1990** [comment]
@ufb, apparently, uninstalling and then re-installing the add-on works OK. I'm not able to reproduce it anymore in my browser, I tried both FF and FF Developers Edition.

**Comment by ufb** [comment]
> apparently, uninstalling and then re-installing the add-on works OK

Yes, for some users this also helped with their last FF update - but for others it did not

**Comment by cheeberger** [comment]
I believe it is worth mentioning that text marker still works perfectly on the latest FF update for my mac pro running mojave.

**Comment by sigfriedCub1990** [comment]
> I believe it is worth mentioning that text marker still works perfectly on the latest FF update for my mac pro running mojave.

I had another weird issue with Mojave, FF and TextMarker, this time I'm unable to see the `marked` elements even after TextMarker itself said `Highlights restored`.

**Comment by zsq2010** [comment]
walking around to see some improving and cheering~~
